R/hiper_glm.R

```diff
@@ -4,7 +4,8 @@ hiper_glm <- function(design, outcome, model_name = "linear", option = list()) {
   if (!(model_name %in% supported_model)) {
     stop(sprintf("The model %s is not supported.", model_name))
   }
-  model <- new_regression_model(design, outcome, model_name)
+  option <- initialize_option(option)
+  model <- new_regression_model(design, outcome, model_name, option$SGD_solver, option$use_matvec_via_transp)
```

Comment:
The constructor `new_regression_model()` doesn't need to know whether you are using SGD or not; it just needs to know whether it should transpose the designmatrix or not. So, don't pass `option$SGD_solver`.

Reply:
No longer passing `option$SGD_solver`. Updated `initialize_option()` function and tests